src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/dns"
+require "ipaddr"
+Yast.import "IP"
+
+module Y2Network
+  module Autoinst
+    # This class is responsible of importing the AutoYast interfaces section
+    class InterfacesReader
+      include Yast::Logger
+
+      # @return [AutoinstProfile::InterfacesSection]
+      attr_reader :section
+
+      # @param section [AutoinstProfile::InterfacesSection]
+      # TODO: read also udev rules
+      def initialize(section)
+        @section = section
+      end
+
+      # Creates a new {ConnectionConfigsCollection} config from the imported profile interfaces
+      # section
+      # @note interfaces will be created automatic from connection configs
+      #
+      # @return [ConnectionConfigsCollection] the imported connections configs
+      def config
+        configs = @section.interfaces.map do |interface_section|
+          log.info "Creating config for interface section: #{interface_section.inspect}"
+          config = create_config(interface_section)
+          config.propose # propose reasonable defaults for not set attributes
+          load_generic(config, interface_section)
+
+          case config
+          when ConnectionConfig::Vlan
+            load_vlan(config, interface_section)
+          when ConnectionConfig::Bridge
+            load_bridge(config, interface_section)
+          when ConnectionConfig::Bonding
+            load_bonding(config, interface_section)
+          when ConnectionConfig::Wireless
+            load_wireless(config, interface_section)
+          end
+
+          log.info "Resulting config: #{config.inspect}"
+          config
+        end
+
+        ConnectionConfigsCollection.new(configs)
+      end
+
+    private
+
+      def create_config(interface_section)
+        # TODO: autoyast backend for type detector?
+        # TODO: TUN/TAP interface missing for autoyast?
+        return ConnectionConfig::Bonding.new if interface_section.bonding_slave0 && !interface_section.bonding_slave0.empty?
```

Comment:
Probably an `if-else` construction would be more readable (and you could get rid of all those `return` keywords).